lib/gitWorkflow.js

```diff
@@ -0,0 +1,187 @@
+  }
+
+  /**
+   * Get name of backup stash
+   *
+   * @param {Object} [options]
+   * @returns {Promise<Object>}
+   */
+  async getBackupStash() {
+    const stashes = await this.execGit(['stash', 'list'])
+    const index = stashes.split('\n').findIndex(line => line.includes(STASH))
+    return `stash@{${index}}`
+  }
+
+  /**
+   * Create backup stashes, one of everything and one of only staged changes
+   * Leves stages files in index for running tasks
+   *
+   * @param {Object} [options]
+   * @returns {Promise<void>}
+   */
+  async stashBackup() {
+    debug('Backing up original state...')
+
+    // Git stash loses metadata about a possible merge mode
+    // Manually check and backup if necessary
+    if (await checkFile(this.mergeHeadFile)) {
+      debug('Detected current merge mode!')
+      debug('Backing up merge state...')
+      await Promise.all([
+        readBufferFromFile(this.mergeHeadFile).then(
+          mergeHead => (this.mergeHeadBuffer = mergeHead)
+        ),
+        readBufferFromFile(this.mergeModeFile).then(
+          mergeMode => (this.mergeModeBuffer = mergeMode)
+        ),
+        readBufferFromFile(this.mergeMsgFile).then(mergeMsg => (this.mergeMsgBuffer = mergeMsg))
+      ])
+      debug('Done backing up merge state!')
+    }
+
+    // Git stash ignores new, untracked files, so back them up separately
+    // Use `git ls-files` to list untracked files:
+    // `--others includes` all untracked files, including ignored files
+    // `--exclude-standard` excludes ignored files, the .git/ dir and so on...
+    let untrackedFiles = await this.execGit(['ls-files', '--others', '--exclude-standard'])
+    if (untrackedFiles) {
+      debug('Detected untracked files:')
+      debug(untrackedFiles)
+      debug('Backing up untracked files...')
+      // Resolve untrackedFiles output into filenames
+      untrackedFiles = untrackedFiles
+        .split('\n')
+        .map(file => normalize(path.resolve(this.cwd, file)))
+      this.untrackedFiles = new Map()
+      const readPromises = []
+      for (const file of untrackedFiles) {
+        // Read all untracked files into buffers, and save them into internal Map
+        readPromises.push(
+          readBufferFromFile(file).then(buffer => this.untrackedFiles.set(file, buffer))
+        )
+      }
+      await Promise.all(readPromises)
+      debug('Done backing up untracked files!')
+    }
+
+    // Get stash of entire original state, including unstaged changes
+    // Keep index so that tasks only work on those files
+    await this.execGit(['stash', 'save', '--quiet', '--include-untracked', '--keep-index', STASH])
+    // Since only staged files are now present, get a diff of unstaged changes
+    // by comparing current index against original stash, but in reverse
+    const original = await this.getBackupStash()
+    this.unstagedDiff = await this.execGit([
+      'diff',
+      '--unified=0',
+      '--no-color',
+      '--no-ext-diff',
+      '-p',
```

Comment:
I'd prefer full attribute names for readability here

Reply:
Will fix.